app/controllers/users_controller.rb

```diff
@@ -11,10 +11,10 @@ def edit
 
   def update
     @user = current_user
-    if update_params[:password] == ""
-      @user.update(update_params.except(:password, :password_confirmation))
-    elsif update_params[:password]
-      @user.update(update_params)
+    if user_info[:password] == ""
+      @user.update(user_info.except(:password))
+    elsif user_info[:password]
+      @user.update(user_info)
```

Comment:
Probably wise to `permit(:password_confirmation)` in the `user_info` method so that BCrypt can handle it appropriately if the user submits a password change and the confirmation field doesn't match. Otherwise, looks good!

Reply:
No problem, went a head and made that adjustment. user info no permits password conf and added additional exception to that line.

All tests pass.